### DISCUSSION TITLE
How to use with nginx?

### DISCUSSION THREAD

**Discussion by failable** [marimo-team/marimo#3144] [Q&A]
I have `/etc/nginx/sites-available/dashboard.my.website`

```
server {                                                                                                                             
    listen 80;                                                                                                                      
    server_name dashboard.my.website;                                                                            
    location / {                                                                                                                    
        proxy_pass http://127.0.0.1:2718;                                                                           
    }                                                                                                                                     
}  
```

and start marimo with 

```
uv run marimo run src/scripts/dashboard/dashboard.py --headless --host 0.0.0.0 --proxy dashboard.my.website:80  
```

I can open the URL but I get `kernel not found` error. How should I fix it? Thanks.

<img width="286" alt="image" src="https://github.com/user-attachments/assets/d013ab47-47b4-4c99-a590-a353c0e8f3c1" />

**Comment by mscolnick** [comment]
Can you provide more error info? anything in the browser console or your server console? marimo uses websockets and you might need a bit more nginx config to support websockets.

**Reply by failable** [reply]
Here is the browser console output

<img width="1919" alt="Screenshot 2024-12-13 at 11 22 26" src="https://github.com/user-attachments/assets/754c37ca-5c41-4c65-913d-8eaa253142d3" />

**Reply by mscolnick** [reply]
yea it looks like it cannot connect. likely ws configuration on the nginx side, unfortunately, i am not too familiar with nginx config these day.
